lambdas/data-migration1/tests/test-index.js

```diff
@@ -0,0 +1,214 @@
+
+const {
+  generateLocalTestDb,
+  destroyLocalTestDb,
+  localStackConnectionEnv,
+} = require('@cumulus/db');
+
+// eslint-disable-next-line node/no-unpublished-require
+const { migrationDir } = require('../../db-migration');
+const { handler } = require('../dist/lambda');
+
+let asyncOperationsModel;
+let collectionsModel;
+let providersModel;
+let rulesModel;
+
+const testDbName = `data_migration_1_${cryptoRandomString({ length: 10 })}`;
+const workflow = cryptoRandomString({ length: 10 });
+
+test.before(async (t) => {
+  process.env.stackName = cryptoRandomString({ length: 10 });
+  process.env.system_bucket = cryptoRandomString({ length: 10 });
+
+  await createBucket(process.env.system_bucket);
+
+  process.env.AsyncOperationsTable = cryptoRandomString({ length: 10 });
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.ProvidersTable = cryptoRandomString({ length: 10 });
+  process.env.RulesTable = cryptoRandomString({ length: 10 });
+
+  const workflowfile = `${process.env.stackName}/workflows/${workflow}.json`;
+  const messageTemplateKey = `${process.env.stackName}/workflow_template.json`;
+
+  const createKeyResponse = await KMS.createKey();
+  process.env.provider_kms_key_id = createKeyResponse.KeyMetadata.KeyId;
+
+  asyncOperationsModel = new AsyncOperation({
+    stackName: process.env.stackName,
+    systemBucket: process.env.system_bucket,
+  });
+  collectionsModel = new Collection();
+  providersModel = new Provider();
+  rulesModel = new Rule();
+
+  await asyncOperationsModel.createTable();
+  await collectionsModel.createTable();
+  await providersModel.createTable();
+  await rulesModel.createTable();
+
+  await Promise.all([
+    putJsonS3Object(
+      process.env.system_bucket,
+      messageTemplateKey,
+      { meta: 'meta' }
+    ),
+    putJsonS3Object(
+      process.env.system_bucket,
+      workflowfile,
+      { testworkflow: 'workflow-config' }
+    ),
+  ]);
+  const { knex, knexAdmin } = await generateLocalTestDb(testDbName, migrationDir);
+  t.context.knex = knex;
+  t.context.knexAdmin = knexAdmin;
+
+  process.env = {
+    ...process.env,
+    ...localStackConnectionEnv,
+    PG_DATABASE: testDbName,
+  };
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex('providers').del();
+  await t.context.knex('collections').del();
+  await t.context.knex('rules').del();
+  await t.context.knex('async_operations').del();
+});
+
+test.after.always(async (t) => {
+  await providersModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await rulesModel.deleteTable();
+  await asyncOperationsModel.deleteTable();
+
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+
+  await destroyLocalTestDb({
+    knex: t.context.knex,
+    knexAdmin: t.context.knexAdmin,
+    testDbName,
+  });
+});
+
+test.only('handler migrates async operations, collections, providers, rules', async (t) => {
+  const fakeCollection = {
```

Comment:
Was there a reason (and explicitness is certainly valid) this and the following aren't using the exported test factories from api/lib/testUtils given we're already bringing the api packages into the migration lambda?

Reply:
Looking more closely at this, to some degree you'd be passing in a lot of overrides, so perhaps that doesn't make sense.    This is solidly in 🐜  territory as we're not likely to change the schema at this point.